src/trivia/libraries/problems/problems.go

```diff
@@ -0,0 +1,58 @@
+package problems
+
+import (
+	"encoding/csv"
+	"flag"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// GetProblems returns trivia questions
+func GetProblems() ([]Problem, error) {
+	csvFilename := flag.String("csv", "problems.csv", "a csv file in the format of 'question,option,option,option,answer'")
+	flag.Parse()
+
+	file, err := os.Open(*csvFilename)
+	if err != nil {
+		return nil, err
+	}
+
+	r := csv.NewReader(file)
+	lines, err := r.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+
+	return parseLines(lines)
+}
+
+func parseLines(lines [][]string) ([]Problem, error) {
+	ret := make([]Problem, len(lines))
+	for i, line := range lines {
+		solutionOptionNumber, err := strconv.ParseInt(strings.TrimSpace(line[4]), 10, 64)
+		if err != nil {
+			return nil, err
+		}
+
+		ret[i] = Problem{
+			Question:           line[0],
+			Option1:            line[1],
+			Option2:            line[2],
+			Option3:            line[3],
```

Comment:
this code in general assumes that line has a certain length and would panic if it gets a line of a different length. Panics are undesirable because it's hard to annotate them with useful information and they don't flow as nicely as errors.

it's worth rewriting this code to be a bit more picky like we would do, so the first thing it does to each line is check to make sure the length is expected and then immediately returns an error if that isn't so.

solutionOptionNumber should also be checked to make sure it's not going to just say get me line[100] and cause a panic